packages/api/authentication/handlers.go

```diff
@@ -0,0 +1,172 @@
+package authentication
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gorilla/sessions"
+	"github.com/jiulongw/siwe-go"
+)
+
+var Store = sessions.NewCookieStore([]byte(os.Getenv("SESSION_KEY")))
+
+// seconds
+const MAX_NONCE_AGE = 30
```

Comment:
Is it necessary to expire the nonce? At a minimum, 30 seconds seems quite aggressive. Maybe 5min? Either way, I don't think there is a security concern with the nonce lifetime being indefinite

Reply:
Another approach we could take, is for the signer to sign the lastest block hash, and then verify that it is within the last 30seconds on the server. That would allow us to remove the nonce endpoint